nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+# Temporal Go SDK - Nexus Handler
+
+## Overview
+
+An addition to the [Temporal Go SDK][temporal-go-sdk] that exposes the ability for users to handle Nexus API requests.
+
+**Goals**
+
+- Expose a simple, high level API for binding Nexus operations to common Temporal constructs
+- Support for starting and canceling operations
+- Support multi-tenant applications with minimal boilerplate
+
+**Non-Goals**
+
+- Support for getting an operation's result or info
+- API definitions for other SDKs
+- API for invoking Nexus calls - refer to the [Nexus Go SDK][nexus-go-sdk] for an out-of-workflow reference client
+- Server implementation details
+- Worker / server API contract
+- Allow advanced users to implement Nexus inbound and outbound request proxies for cross-organization calls where
+  data converters are not shared and data is not exposed to Temporal server.
+
+  This will quickly become a requirement for production use cases, especially with Temporal Cloud's zero trust model. It
+  will be addressed shortly after Nexus becomes available for preview.
+
+> NOTE: These are non-goals for _this_ proposal, but they will be defined in follow up proposals / PRs.
+
+## Background
+
+Nexus is a thin protocol on top of HTTP, defined in [this spec][nexus-http-api]. Temporal serves as an HTTP proxy,
+translating Nexus requests into tasks that can be polled on by an SDK worker on its configured task queue.
+Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker
+to be processed.
+
+The process of translating requests into tasks and delivering them to workers is out of scope for this proposal.
+
+## SDK Additions
+
+The Temporal Go SDK exposes a `Worker` interface that embeds a `Registry` interface defined as:
+
+```go
+type Registry interface {
+	WorkflowRegistry
+	ActivityRegistry
+}
+```
+
+Add and embed a third registry type: `OperationRegistry` with the following method:
+
+```go
+type OperationRegistry interface {
+	RegisterOperation(operation string, handler operation.Handler)
+	// Alternative: get the operation name from handler.Name()
+	// RegisterOperation(handler operation.Handler)
+}
+```
+
+`operation.Handler` is an interface with a minimal definition of:
+
+```go
+package operation // go.temporal.io/sdk/operation
```

Comment:
> go.temporal.io/sdk/operation

This should either be in a Nexus-specific area or have Nexus-specific prefixes IMO

Reply:
I actually like that we don't expose Nexus in the SDK. Operation is a new concept, like an activity or a workflow and deserves a top level namespace IMHO.

Reply:
I think we should expose this as Nexus. Nexus is the new concept, not just Nexus operations. Nexus handlers, calling non-Temporal Nexus servers, etc. Unless you plan on supporting non-Nexus operation forms, there's no value in hiding IMO.

f we _must_ hide that we're using Nexus while also expecting them, I think we need a new word or at least an additional adjective besides "operation" which is fairly overloaded.

Reply:
I don't have super strong opinion here, I like that operations are a top level concept and it's not like we're hiding Nexus completely especially if we're reusing some of the Nexus Go SDK types.
`go.temporal.io/sdk/worker/nexus`?